map.go

```diff
@@ -1400,7 +1401,7 @@ func (m *Map) finalize(spec *MapSpec) error {
 		}
 	}
 
-	if spec.Freeze {
+	if spec.readOnly() {
```

Comment:
I don't think this is correct: it is valid for a map to be BPF_F_RDONLY_PROG but mutable from user space. Your change breaks this behaviour.

Reply:
Ack, will change this to a name-based trigger like libbpf instead.